lib/pbench/agent/tool_meister.py

```diff
@@ -1311,125 +1347,90 @@ def _send_directory(self, directory, uri, ctx):
         target_dir = directory.name
         parent_dir = directory.parent
         tar_file = parent_dir / f"{target_dir}.tar.xz"
-        o_file = parent_dir / f"{target_dir}.tar.out"
-        e_file = parent_dir / f"{target_dir}.tar.err"
+
         try:
-            # Invoke tar directly for efficiency.
-            with o_file.open("w") as ofp, e_file.open("w") as efp:
-                cp = subprocess.run(
-                    [
-                        self.tar_path,
-                        "--create",
-                        "--xz",
-                        "--force-local",
-                        f"--file={tar_file}",
-                        target_dir,
-                    ],
-                    cwd=parent_dir,
-                    stdin=None,
-                    stdout=ofp,
-                    stderr=efp,
+            tar_process = self._create_tar(directory, tar_file)
+            if tar_process.returncode != 0:
+                # Re-try creating tar ball by suppressing all the warnings
+                tar_process = self._create_tar(
+                    directory, tar_file, ignore_warning=["none"]
                 )
+                if tar_process.returncode != 0:
+                    # Tar ball creation failed even after suppressing all the warnings,
+                    # we will now proceed to create an empty tar ball
+                    self._create_tar(
+                        Path("/dev/null"), tar_file, ignore_warning=["none"]
+                    )
```

Comment:
This empty tar file thing is weird. It occurs to me that, if the protocol really requires something to be `PUT` to TDS to continue (it'd be better to just be able to tell the TDS listener that we have nothing to PUT), maybe we should consider storing the tar status and `stderr` stream in a file in the directory so it gets recorded for triage as part of the run?

Reply:
Yeah but we already log through the Redis, so the error message should already be there on the controller in tm.log file